nf_core/module-template/modules/main.nf

```diff
@@ -78,6 +79,9 @@ process {{ tool_name_underscore|upper }} {
         {%- endif %}
         $bam
 
-    echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' > ${software}.version.txt
+    cat <<-END_VERSIONS > versions.yml
+    ${task.process}:
```

Comment:
Just to be sure: is `task.process` what we want here? 
Is it just the process name, or can it be a long concatenation of "WORKFLOW:SUBWORKFLOW:PROCESS"?

Reply:
It's the `WORKFLOW:SUBWORKFLOW:PROCESS`, but if it's sufficient for just `PROCESS`, then we can add `tokenize(':')[-1]` to it, or replace with `software` if the subtool is not important. Then we may also need add the `unique` bit back into the channel processing.

Reply:
Hmm, when you override a container with `process:withName` I think it just uses the last part?

Reply:
True, so then maybe the `PROCESS` is what we want. 
```suggestion
    ${task.process.tokenize(':')[-1]}:
```